packages/next/build/webpack-config.ts

```diff
@@ -61,6 +62,16 @@ const escapePathVariables = (value: any) => {
     : value
 }
 
+const devtoolRevertWarning = execOnce(
+  (devtool: webpack.Configuration.devtool) => {
+    console.warn(
+      chalk.yellow.bold('Warning: ') +
+        chalk.bold(`Reverting webpack devtool to '${devtool}'.\n`) +
+        'Changing the webpack devtool in development mode will cause severe performance regressions.\n'
```

Comment:
This needs an err.sh link that explains that Next.js already chooses the most optional setting for development.
This message should also instruct the user to remove their `devtool` overriding or apply it to production builds only.